File: DSA/Manav-Rajput/Day38_question1.java
class Solution {
    String removeDups(String str) {
        HashSet<Character> seen = new HashSet<>();
       
        StringBuilder result = new StringBuilder();

        for (int i = 0; i < str.length(); i++) {
            char c = str.charAt(i);
            if (!seen.contains(c)) {
                seen.add(c);
                result.append(c);
            }
        }
        
        return result.toString();
    }
}

File: DSA/Arshdeep-Kaur/Day 38/Day38_Question1.cpp
class Solution {
  public:
    string removeDups(string str) {
        int count[26] = {0};
        int i = 0;
        
        while (i < str.length()) {
            if (count[str[i] - 'a'] == 0) {
                count[str[i] - 'a']++;
                i++;
            } else {
                str.erase(str.begin() + i);
            }
        }
        
        return str;
    }
};
